vkcs/resource_vkcs_publicdns_record.go

```diff
@@ -0,0 +1,411 @@
+	client, err := config.PublicDNSV2Client(getRegion(d, config))
+	if err != nil {
+		return diag.Errorf("Error creating VKCS public DNS client: %s", err)
+	}
+
+	zoneID := d.Get("zone_id").(string)
+	recordType := d.Get("type").(string)
+	updateOpts := recordUpdateOpts{
+		Name:     d.Get("name").(string),
+		IPv4:     d.Get("ip_v4").(string),
+		IPv6:     d.Get("ip_v6").(string),
+		Content:  d.Get("content").(string),
+		Priority: d.Get("priority").(int),
+		Weight:   d.Get("weight").(int),
+		Host:     d.Get("host").(string),
+		Port:     d.Get("port").(int),
+		TTL:      d.Get("ttl").(int),
+	}
+
+	log.Printf("[DEBUG] vkcs_publicdns_record update options: zone_id: %s, type: %s, opts: %#v",
+		zoneID, recordType, updateOpts)
+
+	_, err = recordUpdate(client, zoneID, d.Id(), updateOpts, recordType).Extract()
+	if err != nil {
+		return diag.FromErr(checkDeleted(d, err, "Error updating vkcs_publicdns_record"))
+	}
+
+	return resourcePublicDNSRecordRead(ctx, d, meta)
+}
+
+func resourcePublicDNSRecordDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(configer)
+	client, err := config.PublicDNSV2Client(getRegion(d, config))
+	if err != nil {
+		return diag.Errorf("Error creating VKCS public DNS client: %s", err)
+	}
+
+	zoneID := d.Get("zone_id").(string)
+	recordType := d.Get("type").(string)
+
+	err = recordDelete(client, zoneID, d.Id(), recordType).ExtractErr()
+	if err != nil {
+		return diag.FromErr(checkDeleted(d, err,
+			fmt.Sprintf("Error deleting vkcs_publicdns_record: zone_id: %s, type: %s, id:", zoneID, recordType)))
+	}
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{recordStatusActive},
+		Target:     []string{recordStatusDeleted},
+		Refresh:    publicDNSRecordStateRefreshFunc(client, zoneID, d.Id(), recordType),
+		Timeout:    d.Timeout(schema.TimeoutCreate),
+		Delay:      recordDelay,
+		MinTimeout: recordMinTimeout,
+	}
+
+	_, err = stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return diag.Errorf("Error waiting for vkcs_publicdns_record %s to become deleted: %s", d.Id(), err)
+	}
+
+	return nil
+}
+
+func resourcePublicDNSRecordCustomizeDiff(ctx context.Context, diff *schema.ResourceDiff, v interface{}) error {
+	recordType := diff.Get("type").(string)
+
+	if recordType == recordTypeSRV {
+		diff.SetNew("name", extractPublicDNSRecordSRVName(diff))
+	}
+
+	args := getPublicDNSRecordArgs(recordType)
+
+	var changedArgs []string
+	for k := range diff.GetRawPlan().AsValueMap() {
+		if diff.HasChange(k) {
+			changedArgs = append(changedArgs, k)
+		}
+	}
+
+OuterLoop:
+	for _, k := range changedArgs {
+		for _, ak := range args {
+			if k == ak {
+				continue OuterLoop
+			}
+		}
+		return fmt.Errorf("\"%s\" is not supported for records of type \"%s\", supported args are: %v", k, recordType, args)
+	}
+
+	return nil
+}
+
+func resourcePublicDNSRecordDescription() string {
+	argsJSON, _ := json.MarshalIndent(getPublicDNSRecordArgsMap(), "", "  ")
```

Comment:
we render argument map as json?

Reply:
md table would be much better

Reply:
Updated

Reply:
have you checked the result with doc generator?